llvm/unittests/ADT/STLForwardCompatTest.cpp

```diff
@@ -119,4 +119,20 @@ TEST(TransformTest, MoveTransformLlvm) {
   EXPECT_EQ(0u, MoveOnly::Destructions);
 }
 
+TEST(TransformTest, ToUnderlying) {
+  enum E { A1 = 0, B1 = -1 };
+  static_assert(llvm::to_underlying(A1) == 0);
+  static_assert(llvm::to_underlying(B1) == -1);
+
+  enum E2 : unsigned char { A2 = 0, B2 };
+  static_assert(std::is_same_v<unsigned char, decltype(llvm::to_underlying(A2))>);
+  static_assert(llvm::to_underlying(A2) == 0);
+  static_assert(llvm::to_underlying(B2) == 1);
+
+  enum class E3 { A3 = -1, B3 };
+  static_assert(std::is_same_v<int, decltype(llvm::to_underlying(E3::A3))>);
```

Comment:
Is this guaranteed to be an `int`? Maybe specify the `enum` type to make this clear -- it won't change test coverage.

Reply:
See https://timsong-cpp.github.io/cppwp/n4659/dcl.enum#7

Reply:
https://eel.is/c++draft/enum#dcl.enum-5.sentence-4

Reply:
Clause 5 above states that *For a scoped enumeration type, the underlying type is int if it is not explicitly specified.*